common/src/auth/ServiceConnectionString.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text;
+
+namespace Microsoft.Azure.Devices.Common.Authorization
+{
+    /// <summary>
+    /// This object handles the connection string for the Azure IoT Services.
+    /// </summary>
+    /// <remarks>
+    /// The connection string contains a set of information that uniquely identify an IoT Service.
+    ///
+    /// A valid connection string shall be in one of the following formats:
+    /// <code>
+    /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessKey=[Key]
+    /// HostName=[ServiceName];SharedAccessKeyName=[keyName];SharedAccessSignature=[Signature]
+    /// </code>
+    ///
+    /// This object parse and store the connection string. It is responsible to provide the authorization token too.
+    /// </remarks>
+    public class ServiceConnectionString
+    {
+        protected static readonly TimeSpan DefaultTokenTimeToLive = TimeSpan.FromMinutes(5);
+
+        /// <summary>
+        /// CONSTRUCOR
```

Comment:
I've noticed we do this, but I wonder why. "CONSTRUCTOR" doesn't seem very helpful, especially when objects have several ctor overrides.

In this case, how about "Creates an instance based on a supplied parser"?

Reply:
changed it.